app/org/sagebionetworks/bridge/dynamodb/DynamoUploadDedupeDao.java

```diff
@@ -0,0 +1,63 @@
+package org.sagebionetworks.bridge.dynamodb;
+
+import javax.annotation.Resource;
+
+import com.amazonaws.services.dynamodbv2.datamodeling.DynamoDBMapper;
+import com.amazonaws.services.dynamodbv2.datamodeling.DynamoDBQueryExpression;
+import com.amazonaws.services.dynamodbv2.model.AttributeValue;
+import com.amazonaws.services.dynamodbv2.model.ComparisonOperator;
+import com.amazonaws.services.dynamodbv2.model.Condition;
+import org.joda.time.LocalDate;
+import org.springframework.stereotype.Component;
+
+import org.sagebionetworks.bridge.BridgeConstants;
+import org.sagebionetworks.bridge.dao.UploadDedupeDao;
+import org.sagebionetworks.bridge.schema.UploadSchemaKey;
+
+/** DDB implementation of UploadDedupeDao. */
+@Component
+public class DynamoUploadDedupeDao implements UploadDedupeDao {
+    private static final int CREATED_ON_DELTA_MILLIS = 1000;
+
+    private DynamoDBMapper mapper;
+
+    /** UploadDedupe DDB mapper. */
+    @Resource(name = "uploadDedupeDdbMapper")
+    public void setMapper(DynamoDBMapper mapper) {
+        this.mapper = mapper;
+    }
+
+    /** {@inheritDoc} */
+    @Override
+    public boolean isDuplicate(long createdOn, String healthCode, UploadSchemaKey schemaKey) {
+        // Hash key comes from health code and schema
+        DynamoUploadDedupe hashKey = new DynamoUploadDedupe();
+        hashKey.setHealthCode(healthCode);
+        hashKey.setSchemaKey(schemaKey.toString());
+
+        // range key, anything within 1 second is valid
+        Condition createdOnCondition = new Condition().withComparisonOperator(ComparisonOperator.BETWEEN)
+                .withAttributeValueList(
+                        new AttributeValue().withN(String.valueOf(createdOn - CREATED_ON_DELTA_MILLIS)),
+                        new AttributeValue().withN(String.valueOf(createdOn + CREATED_ON_DELTA_MILLIS)));
```

Comment:
I'm surprised duplicates can have different createdOn timestamps... we really have no other way to determine if it's a previous upload? MD5 or something? It'll never be the case that a client sends files with createdOn values that are close to one another that are nevertheless different content?